client.go

```diff
@@ -27,21 +27,13 @@ type AutoNATError struct {
 	Text   string
 }
 
-// GetAddrs is a function that returns the addresses to dial back
-type GetAddrs func() []ma.Multiaddr
-
 // NewAutoNATClient creates a fresh instance of an AutoNATClient
-// If getAddrs is nil, h.Addrs will be used
-func NewAutoNATClient(h host.Host, getAddrs GetAddrs) AutoNATClient {
-	if getAddrs == nil {
-		getAddrs = h.Addrs
-	}
-	return &client{h: h, getAddrs: getAddrs}
+func NewAutoNATClient(h host.Host) AutoNATClient {
```

Comment:
Note: potentially breaking an externally-visible signature. should allow consumers to transition off before removing.

Reply:
This is not just a signature issue, it's a core functionality issue. Clients may rewrite the addresses returned by the host.